VocaDb.Migrations/Migrations.cs

```diff
@@ -1,10 +1,149 @@
+			};
+
+			// insert into Comments(OldTable, OldId, Author, Created, Message)
+			//     select 'AlbumComments', Id, Author, Created, Message from AlbumComments union
+			//     select 'ArtistComments', Id, Author, Created, Message from ArtistComments union
+			//     select 'discussions.DiscussionComments', Id, Author, Created, Message from discussions.DiscussionComments union
+			//     ...
+			//     select 'UserComments', Id, Author, Created, Message from UserComments union
+			//     select 'DiscussionTopics', Id, Author, Created, Message from discussions.DiscussionTopics union
+			//     select 'AlbumReviews', Id, Author, Created, Message from AlbumReviews
+			// order by Created
+			var values = string.Join(" union ", commentTables.Select(table => $"select '{table.NameWithSchema}', Id, Author, Created, Message from {table.NameWithSchema}"));
+			Execute.Sql($@"insert into {TableNames.Comments}(OldTable, OldId, Author, Created, Message) {values} order by Created");
+
+			foreach (var table in commentTables)
+			{
+				// Add `Comment` column to comment tables (`AlbumComments`, `ArtistComments`, `DiscussionComments`, ..., 'DiscussionTopics', 'AlbumReviews').
+				Create.Column("Comment").OnTable(table.Name).InSchema(table.Schema).AsInt32().Nullable();
+				// e.g. update discussions.DiscussionComments set Comment = c.Id from discussions.DiscussionComments ec inner join Comments c on c.OldTable = 'discussions.DiscussionComments' and ec.Id = c.OldId
+				Execute.Sql($"update {table.NameWithSchema} set Comment = c.Id from {table.NameWithSchema} ec inner join {TableNames.Comments} c on c.OldTable = '{table.NameWithSchema}' and ec.Id = c.OldId");
+				Alter.Column("Comment").OnTable(table.Name).InSchema(table.Schema).AsInt32().NotNullable().ForeignKey(TableNames.Comments, "Id");
+
+				// TODO: remove these columns.
+
+				// Make `Author` nullable and rename it to `OldAuthor`.
+				Alter.Column("Author").OnTable(table.Name).InSchema(table.Schema).AsInt32().Nullable();
+				Rename.Column("Author").OnTable(table.Name).InSchema(table.Schema).To("OldAuthor");
+
+				// Make `Created` nullable and rename it to `OldCreated`.
+				Alter.Column("Created").OnTable(table.Name).InSchema(table.Schema).AsDateTime().Nullable();
+				Rename.Column("Created").OnTable(table.Name).InSchema(table.Schema).To("OldCreated");
+
+				// Make `Message` nullable and rename it to `OldMessage`.
+				Alter.Column("Message").OnTable(table.Name).InSchema(table.Schema).AsString(4000).Nullable();
+				Rename.Column("Message").OnTable(table.Name).InSchema(table.Schema).To("OldMessage");
+
+				if (Schema.Schema(table.Schema).Table(table.Name).Column("AuthorName").Exists())
+				{
+					// Make `AuthorName` nullable and rename it to `OldAuthorName`.
+					Alter.Column("AuthorName").OnTable(table.Name).InSchema(table.Schema).AsString(100).Nullable();
+					Rename.Column("AuthorName").OnTable(table.Name).InSchema(table.Schema).To("OldAuthorName");
+				}

```

Comment:
Why do `AlbumComments` and `DiscussionComments` tables have the `AuthorName` column? Can we remove it?

Reply:
Originally there was an idea that you could delete the user account (remove it from database), making the Author field NULL, but preserving the name of the user who posted the comment. But I scrapped that idea at some point, it's better to just keep user references non-nullable and clear user information if needed, never deleting the user entity.

I guess there is still some value keeping record of the name that user had when posting the comment (users can change their username), but I don't see it as necessary.